lib/gateway/request.js

```diff
@@ -1,54 +1,40 @@
 'use strict'
-const http = require('http')
-const https = require('https')
+const { Pool } = require('undici')
+const { URL } = require('url')
 const eos = require('end-of-stream')
 const { FederatedError } = require('../errors')
 
 function agentOption (opts) {
   return {
     keepAlive: true,
-    keepAliveMsecs: opts.keepAliveMsecs || 60 * 1000, // 1 minute
-    maxSockets: opts.maxSockets || 2048,
-    maxFreeSockets: opts.maxFreeSockets || 2048,
-    rejectUnauthorized: opts.rejectUnauthorized
+    keepAliveMaxTimeout: opts.keepAliveMsecs || 60 * 1000, // 1 minute
+    connections: opts.connections || 10
```

Comment:
```suggestion
    connections: opts.connections || opts.maxSockets || 10
```

Backward compatibility